brave/brave-core
# Keyring pref

Resolves https://github.com/brave/brave-browser/issues/17137

This PR 
1. Separates keyring pref by keyring id (we only use `default` currently)
2. Simplifies AddAccount process and stored account name in account metas keyed by account path
3. Makes `HDKeyring` only accessible by `KeyringController`. Adding an account must go through `KeyringController` first because we store extra account metas in it. In the future, we can add `ModifyAccountName` and add more meta to it if needed.
4. Migrates old prefs to new format

## Submitter Checklist:


## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:
1. Launch Brave without this PR with fresh profile
2. Enable brave://flags/#native-brave-wallet
3. Create or restore wallet
4. Create 5 accounts with customized names
5. Quit Brave and relaunch Brave with this PR
6. There should be exactly same amount of accounts with customized names
